src/Utilities/Compiler/WellKnownTypeProvider.cs

```diff
@@ -20,6 +24,15 @@ private WellKnownTypeProvider(Compilation compilation)
         {
             Compilation = compilation;
             _fullNameToTypeMap = new ConcurrentDictionary<string, INamedTypeSymbol?>(StringComparer.Ordinal);
+            _referencedAssemblies = new Lazy<ImmutableHashSet<IAssemblySymbol>>(
+                () =>
+                {
+                    return ImmutableHashSet.Create<IAssemblySymbol>(
```

Comment:
I was thinking, can different modules in the same compilation reference the same assembly? That might have made the original fallback logic not work, since we'd visit the same assembly more than once.

Reply:
It's an interesting point, but I think in most cases a compilation only has one module.

Reply:
I guess this still might have the same issue if we get back different IAssemblySymbol objects for the same assembly for some reason, but I'm not going to worry about it. 😄